### PR TITLE
Update Pigweed from b227355 to 0aeedcc

### PR DESCRIPTION
**Pull request by mohrr**
Update the Pigweed submodule

Also add script to simplify updating the Pigweed submodule. Script is not perfect but should help next time.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Comment by woody-apple** [comment]
@mohrr Thank you for the update! Can you please restyle as per: https://github.com/project-chip/connectedhomeip/pull/6382

**Comment by andy31415** [comment]
Marking as hotfix: this pigweed build fixes compiling issues after mac updates (saw at least Martin and Jerry have this issue).

ZAP template error seems wrong, so will ignore it. Waiting for the rest of the CI before merging.

**Pull request merged**
